### PR TITLE
Run action fast

### LINKED ISSUE #64: Publish docker image please!
Hi, i have to build this action using the dockerfile every run which takes 30 seconds. Can you just publish the docker image instead? It would speed everyones CI up by 20-25 seconds every run :)

Thanks!

### PR DESCRIPTION
**Pull request by itsmelion**
**Motivation:**
Build time, and time is money. (ye.. they charge for build time, haha) and is just not fair a slack message consume so much  in the build process.

Targets: **2.x**
fixes #64 

### What's in here:
- Skip docker build (use native node from GitHub environment);
- Build the action with [@vercel/ncc](https://github.com/vercel/ncc) as recommended in [Actions docs](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github) This way we don't conflict with the host Repo, and don't even need to waste time installing our own deps;
- Handle errors, and such with Github SDK;
- One less dependency (yargs) once we parse the 'args' as input (with Github SDK);
- Of course added more Lint;

### CODE REVIEW

**Comment by Ilshidur** [comment]
Hello Christopher. I currently don't have a lot of time those days to get done with my TODOs, so this PR is very welcome.
This looks **really** nice, by the way ! I'll review this very soon. Thanks for the work. I hope people will like it.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -4,40 +4,34 @@ on:
...
-      - name: Slack notification FAILED
-        uses: Ilshidur/action-slack@master
-        if: failure()
-        env:
-          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
-          SLACK_AVATAR: sender # optional: repository, sender or ImageURL
-        with:
-          args: "Lint checks *FAILED* \nfrom {{ GITHUB_ACTOR }} at {{ GITHUB_REPOSITORY }}"
+    - name: Slack notification FAILED
+      uses: Ilshidur/action-slack@v2
```
Comment (Ilshidur) [inline .github/workflows/main.yml:26]: **Personal TODO :** update this once I cut a new release with these changes. I'll release it under the `2.1.0` version.

File: entrypoint.js [hunk 2]
```diff
@@ -1,28 +1,36 @@
...
+        process.exitCode = 0;
+        return core.info('Message sent! Shutting down ...');
+      })
+      .catch((err) => {
+        process.exitCode = 1;
+        const errMessage = err.response ? err.response.data : err.message;
+        return core.setFailed(`Error: ${errMessage}`);
+      })
+  }
+})();
```
Comment (Ilshidur) [inline entrypoint.js:36]:
```suggestion
if (!process.exitCode) {
  core.info("Sending message ...");

  axios.post(process.env.SLACK_WEBHOOK, message.get())
    .then(() => {
      process.exitCode = 0;
      return core.info('Message sent! Shutting down ...');
    })
    .catch((err) => {
      process.exitCode = 1;
      const errMessage = err.response ? err.response.data : err.message;
      return core.setFailed(`Error: ${errMessage}`);
    })
}
```

I guess we can remove this `(() => { /* code... */ })()` thing as I primarily used it to make the POST call with a top-level `async`/`await` handling.

**Review by Ilshidur — CHANGES_REQUESTED** [review]
Ok so this was about time for a review (sorry for the waiting :smile:).
This looks perfect. I'll try to test it further tomorrow. In the meantime, feel free let me know what you think of my tiny suggestions.

I really appreciate your PR. Thank you for this work. As you've been clearly improving the quality of this repo, would you like to join as a collaborator ?

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+
```
[the remark this answers is not in the document]
Comment (itsmelion) [reply .github/dependabot.yml]: I guess vulnerability checks for Dependabot need to be activated in the `> Security` tab ([see where](https://docs.github.com/en/github/managing-security-vulnerabilities/configuring-github-dependabot-security-updates#managing-github-dependabot-security-updates-for-your-repositories)), here is just for updates. 

Ye, I've been experimenting both Dependabot and renovate, and i like dependabot, seems faster to respond to changes in the repo that renovate.
Renovate so far has more options to configure, so on my repo's ive been using both (and dealing with the noise haha)
With both active they will be redundant most times, but it's all fine.

**Comment by itsmelion** [comment]
Changes pushed!
@Ilshidur That's great! yea.. (i've been using it alot as well).
all good to me. I will be glad to help as collaborator.

**Comment by Ilshidur** [comment]
Ok, so I tested it and it works as expected. Merging...

I also added you to the repository. Thanks !

**Review by Ilshidur — APPROVED** [review]
All good.

**Pull request merged**